test/sensitivities/linalg/blas.jl

```diff
@@ -0,0 +1,222 @@
+@testset "sensitivities/blas" begin
+
+let ϵ_abs = 1e-5, ϵ_rel = 1e-4, δ = 1e-6
+
+    # Testing allocating sensitivities for dot.
+    import Base.BLAS.dot
+    let rng = MersenneTwister(123456)
+        for _ in 1:10
+            x, y, vx, vy = randn.(rng, [5, 5, 5, 5])
+            δ_abs_x, δ_rel_x = check_Dv(dot, randn(rng), (x, y), δ .* (vx, vy))
+            @test δ_abs_x < ϵ_abs && δ_rel_x < ϵ_rel
```

Comment:
I'm not sure if this is the right way to test for approximate numerical equality. For nonzero numbers, the absolute error will depend on the magnitude of the derivatives, so `δ_abs_x < ϵ_abs` might fail simply because the result is too large. On the other hand, for numbers close to zero, the relative error is ill-defined, which may cause `δ_rel_x < ϵ_rel` to fail.

Given a relative "tolerance factor" `ĉ_rel` and a relative error tolerance `ε̂_abs`, we require that `abs(x - y)` is smaller than `ĉ_rel * eps() * max(abs(x), abs(y))` if the result is nonzero, and that `abs(x - y)` is smaller than `ε̂_abs` if the result is zero. Furthermore, for nonzero numbers, assuming that `ε̂_rel` is sufficiently small, `ĉ_rel * eps() * max(abs(x), abs(y))` will be much smaller than `ε̂_abs`, and for numbers close to zero, `ĉ_rel * eps() * max(abs(x), abs(y))` will be much smaller than `ε̂_abs`; therefore, if `abs(x - y) < max(ε̂_abs, ĉ_rel * eps() * max(abs(x), abs(y)))`, then `x` and `y` should be approximately numerically equal. `isapprox` employs a similar approach, but uses `sum` instead of `max`.